5.0 Draft/IDD/IDDs Authorization System/eu.arrowhead.authorization-http-json.yml

```diff
@@ -0,0 +1,366 @@
+    than this one.
+
+    ## Caching and Performance
+
+    Implementations of this service _may_ allow for its consumers to cache its authorization
+    decisions for brief periods of time when relevant to improve performance. When it is enabled, it
+    is communicated via the `Cache-Control` header of its responses.
+
+    An individual implementation of this service with sufficiently sophisticated software and
+    hardware should be able to handle up to and beyond hundreds of thousands of authorization
+    triplets per second. That being said, there could be cases where no available hardware is
+    performant enough, or where individual consumers are disconnected from this service for
+    significant periods of time. These use cases are outside the scope of this service.
+
+    ## Compression and Language
+
+    An implementation of this service interface _should_ be designed to support compression and
+    _may_ provide human-readable error texts in different languages than English, as described in
+    [RFC 9110, Section 12](https://www.rfc-editor.org/rfc/rfc9110#name-content-negotiation). Not
+    using compression _must_ be supported and _should_ be the default. Also, the default language
+    for error messages _must_ be American English (`en-US`).
+
+    ## Size Limits
+
+    As a mitigation against denial-of-service attacks, all implementations of this service
+    interface _should_ reject incoming requests that are larger than a predefined limit. That limit
+    _must not_ be smaller than 8192 bytes for each received request. An implementation _may_ also
+    limit the sizes of individual parts within each request. It _must_, however, receive request
+    payloads up to 4096 bytes in size. If payload compression is supported, then this limit applies
+    after decompression.
+  version: 5.0.0-rc.1
+  
+  # Additional information present in the IDD template
+  x-transfer-protocol: [HTTP, HTTPS]
+  x-encryption-method: to-add
+  x-encoding: JSON
+  x-compression: to-add
+  x-semantics: to-add
+  x-Ontology: to-add
+
+paths:
+
+  /echo:
+    get:  
+      summary: Endpoint to check the core system availability. 
+      description: Returns an OK to indicate that the core system is available. 
+      responses:
+        '200':
+          description: OK
+
+  /authorizations:
+  post:
+    summary: Bulk Authorize access rights for Resources
+    description: |
+      Allows for bulk authorization requests to be sent to the authorization system in a single API request
+
+      Queries the authorization system for authorization decision for each specified access right over the 
+      specified consumer and resource pair. 
+
+      As a reponse, for each request in the request body, an authorization decision will be returned.
+    requestBody:
+      required: true
+      content:
+        application/json:
+          schema:
+            type: array
+            items:
+              type: object
```

Comment:
We should move this type out into a separate type in the `components` section. The reason for this is that it helps any code generator processing this specification choose a good name for this type. Since the `Triplet` type further down the specification is no longer in use, you could replace it with this type.